lib/model.js

```diff
@@ -3747,6 +3747,38 @@ class Model {
   }
 
   /**
+   * Helper method to determine if a instance is "soft deleted".  This is
+   * particularly useful if the implementer renamed the `deletedAt` attribute
+   * to something different.  This method requires `paranoid` to be enabled.
+   *
+   * @returns {Boolean}
+   */
+  isSoftDeleted() {
+    if (!this.constructor._timestampAttributes.deletedAt) {
+      throw new Error('Model is not paranoid');
+    }
+
+    const deletedAtAttribute = this.constructor.rawAttributes[this.constructor._timestampAttributes.deletedAt];
+    const defaultValue = deletedAtAttribute.hasOwnProperty('defaultValue') ? deletedAtAttribute.defaultValue : null;
+    const deletedAt = this.get(this.constructor._timestampAttributes.deletedAt);
+    const isNotSet = deletedAt === defaultValue;
```

Comment:
Avoid booleans with negatives, it's hard to read something like `!isNotSet`. Prefer `isSet`